### PR TITLE
Integrate featurizers

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**: 
Integrate Featurizer library into ORT build tree but retain it as independently build able.

Make source code compile using MSVC++, make unit tests build and run using GTest.
Create DateTimeTransformer kernel that outputs a non-standard type and register it as Opaque.
Create a unit test and make sure it runs by making necessary adjustments in out test infrastructure.

**Motivation and Context**
We are hosting shared code that will be used by both ORT and ML.NET for AutoML purposes.

### CODE REVIEW

File: onnxruntime/core/automl/featurizers/src/FeaturizerPrep/Featurizer.h [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <memory>
+#include <tuple>
+
+namespace Microsoft {
+namespace Featurizer {
```
Comment (jignparm) [inline onnxruntime/core/automl/featurizers/src/FeaturizerPrep/Featurizer.h]: Featurizer sounds too specific for a namespace. Maybe **AutoML** instead?

Comment (yuslepukhin) [reply onnxruntime/core/automl/featurizers/src/FeaturizerPrep/Featurizer.h]: This is not our code. AutoML team. This code will be shared with them but hosted by ORT.

---
In reply to: [311298415](https://github.com/microsoft/onnxruntime/pull/1573#discussion_r311298415) [](ancestors = 311298415)

**Comment by yuslepukhin** [comment]
@davidbrownellWork is added to the review. #Closed

File: onnxruntime/automl_ops/cpu/datetime_transformer.cc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  explicit DateTimeTransformer(const OpKernelInfo& info) : OpKernel(info) {}
+  Status Compute(OpKernelContext* context) const override;
+};
+
+Status DateTimeTransformer::Compute(OpKernelContext* ctx) const {
+  Status s;
+  auto input_tensor = ctx->Input<Tensor>(0);
+  dtf::TimePoint* output = ctx->Output<dtf::TimePoint>(0);
+
+  int64_t tp = *input_tensor->Data<int64_t>();
```
Comment (gramalingam) [inline onnxruntime/automl_ops/cpu/datetime_transformer.cc]: Is this supposed to be a transformer that transforms a tensor (not necessarily a scalar) of ints to a tensor/array of time_points? Or, is it supposed to be restricted to the case where the input is a scalar? Assuming that automl needs to work with batches of inputs, I would have expected support for a tensor of ints, not just a scalar.

File: onnxruntime/test/automl_ops/datetimetransformer_test.cc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+namespace onnxruntime {
+namespace test {
+
+TEST(DateTimeFeaturizer_DateTime, Past_1976_Nov_17__12_27_04) {
+
+  const time_t date = 217081624;
+  OpTester test("DateTimeTransformer", 1, onnxruntime::kMSAutoMLDomain);
+
+  // We are adding a scalar Tensor in this instance
+  test.AddInput<int64_t>("X", {1}, {date});
```
Comment (gramalingam) [inline onnxruntime/test/automl_ops/datetimetransformer_test.cc]: Is {1} the shape? If so, I think it is better to use the empty shape {} for a scalar (which is the recommended encoding).

File: onnxruntime/test/automl_ops/datetimetransformer_test.cc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  test.AddOutput<dft::TimePoint>("Y", std::move(tp));
+  test.Run(OpTester::ExpectResult::kExpectSuccess);
+}
+
+TEST(DateTimeFeaturizer_Transformer, Past_1976_Nov_17__12_27_05) {
+  const time_t date = 32445842582;
+
+  OpTester test("DateTimeTransformer", 1, onnxruntime::kMSAutoMLDomain);
+  // We are adding a scalar Tensor in this instance
+  test.AddInput<int64_t>("X", {1}, {date});
```
Comment (gramalingam) [inline onnxruntime/test/automl_ops/datetimetransformer_test.cc]: Same comment as above. The {1} should probably be {}, unless we want to support non-scalar inputs also.

File: onnxruntime/automl_ops/cpu/datetime_transformer.cc (hunk 2 above)
Comment (yuslepukhin) [reply onnxruntime/automl_ops/cpu/datetime_transformer.cc]: The answer to this is no. This transformer transforms a single timepoint into a Featurizer specific type. We may come to batches. This kernel is very experimental to verify that we can share the code, create models on a .MET side and run them.

**Pull request merged**